consensus/misc/kip71.go

```diff
@@ -0,0 +1,74 @@
+package misc
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/math"
+	"github.com/klaytn/klaytn/params"
+)
+
+func VerifyKIP71Header(config *params.ChainConfig, parentHeader, header *types.Header) error {
+	if header.BaseFee == nil {
+		return fmt.Errorf("header is missing baseFee")
+	}
+	// Verify the baseFee is correct based on the parent header.
+	expectedBaseFee := CalcBaseFee(parentHeader, config)
+	if header.BaseFee.Cmp(expectedBaseFee) != 0 {
+		return fmt.Errorf("invalid baseFee: have %s, want %s, parentBaseFee %s, parentGasUsed %d",
+			header.BaseFee, expectedBaseFee, parentHeader.BaseFee, parentHeader.GasUsed)
+	}
+	return nil
+}
+
+func CalcBaseFee(parentHeader *types.Header, config *params.ChainConfig) *big.Int {
```

Comment:
Why the `parentGasUsed` is used, not the current block's `gasUsed`? Because the base `gasUsed` is taken from `current block number - 1`, the network usage on `current block number - 1` is considered to calculate the current blocks' base fee.

Reply:
I think It can be confused a little bit. The gasUsed of current block cannot be determined until all the transactions process. Therefore we can also denote parentGasUsed as recentGasUsed.

Reply:
Uh oh, stupid question. Thank you.